src/addr.rs

```diff
@@ -403,6 +403,96 @@ impl Step for VirtAddr {
     }
 }
 
+impl TryFrom<u64> for VirtAddr {
+    type Error = VirtAddrNotValid;
+
+    #[inline]
+    fn try_from(addr: u64) -> Result<Self, Self::Error> {
+        Self::try_new(addr)
+    }
+}
+
+// if `target_pointer_width > 64`, we would need a different error type
+#[cfg(any(target_pointer_width = "32", target_pointer_width = "64"))]
+impl TryFrom<usize> for VirtAddr {
+    type Error = VirtAddrNotValid;
+
+    #[inline]
+    fn try_from(addr: usize) -> Result<Self, Self::Error> {
+        Self::try_new(addr.try_into().unwrap())
+    }
+}
+
+// if `target_pointer_width > 64`, we would need a different error type
+// (and `as` would truncate)
+#[cfg(any(target_pointer_width = "32", target_pointer_width = "64"))]
+impl<T> TryFrom<*const T> for VirtAddr {
+    type Error = VirtAddrNotValid;
+
+    #[inline]
+    fn try_from(addr: *const T) -> Result<Self, Self::Error> {
+        Self::try_new(addr as u64)
+    }
+}
+
+// if target_pointer_width is 128, we would need a different error type
+// (and `as` would truncate)
+#[cfg(any(target_pointer_width = "32", target_pointer_width = "64"))]
+impl<T> TryFrom<*mut T> for VirtAddr {
+    type Error = VirtAddrNotValid;
+
+    #[inline]
+    fn try_from(addr: *mut T) -> Result<Self, Self::Error> {
+        Self::try_new(addr as u64)
+    }
+}
+
+impl From<u32> for VirtAddr {
+    #[inline]
+    fn from(addr: u32) -> Self {
+        Self::new(addr.into())
+    }
+}
+
+#[cfg(any(target_pointer_width = "32", target_pointer_width = "64"))]
+impl<T> From<&T> for VirtAddr {
+    #[inline]
+    fn from(addr: &T) -> Self {
+        Self::new(addr as *const _ as u64)
+    }
+}
+
+#[cfg(any(target_pointer_width = "32", target_pointer_width = "64"))]
+impl<T> From<&mut T> for VirtAddr {
+    #[inline]
+    fn from(addr: &mut T) -> Self {
+        Self::new(addr as *mut _ as u64)
+    }
+}
```

Comment:
These could fail on different architectures. Do we want to care about that given that most users will probably run on x86?

Reply:
Ah yeah, good point! I guess we could a `target_arch` `cfg`-gate to avoid this?

Reply:
Yes, that'd work

Reply:
Changes these to `target_arch`. The `target_pointer_width` ought to be reasonable if the `target_arch` is `x86` or `x86_64`, so I just removed those.

Reply:
I was also wondering if we ought to use `target_arch` for the raw pointer conversions. I left them for now, since they implement `TryFrom` instead of `From`, but those really don't make a whole lot of sense on non-x86 based architectures.